stock_card_report/wizard/stock_card_report_wizard.py

```diff
@@ -0,0 +1,134 @@
+    _description = 'Stock Card Report Wizard'
+
+    date_from = fields.Date(
+        string="Start Date",
+        required=True,
+    )
+    date_to = fields.Date(
+        string="End Date",
+        required=True,
+    )
+    product_id = fields.Many2one(
+        comodel_name='product.product',
+        string='Product',
+        required=True,
+    )
+    stock_location_id = fields.Many2one(
+        comodel_name="stock.location",
+        string="Location",
+        required=True,
+    )
+    initial_stock = fields.Float(
+        string='Initial Stock',
+    )
+    line_ids = fields.One2many(
+        comodel_name='stock.card.report.wizard.line',
+        inverse_name='explosion_id',
+    )
+
+    # @api.onchange('product_id')
+    # def _onchange_product_id(self):
+    #     initial_stock = self.env['stock.quant'].search([
+    #         ('product_id', '=', self.product_id.id)])
+    #     self.initial_stock = initial_stock.quantity
+
+    @api.model
+    def _prepare_line(self, stock_card, type, balance, location):
+        if type == 'in':
+            return {
+                'explosion_id': self.id,
+                'product_id': stock_card.product_id.id,
+                'location_id': stock_card.location_id.id,
+                'location_dest_id': stock_card.location_dest_id.id,
+                'stock_location_id': location,
+                'date': stock_card.date,
+                'reference': stock_card.reference,
+                'product_in': stock_card.product_uom_qty,
+                'product_out': 0.00,
+                'product_balance': balance,
+            }
+        elif type == 'out':
+            return {
+                'explosion_id': self.id,
+                'product_id': stock_card.product_id.id,
+                'location_id': stock_card.location_id.id,
+                'location_dest_id': stock_card.location_dest_id.id,
+                'stock_location_id': location,
+                'date': stock_card.date,
+                'reference': stock_card.reference,
+                'product_in': 0.00,
+                'product_out': stock_card.product_uom_qty,
+                'product_balance': balance,
+            }
+
+    @api.multi
+    def do_explode(self):
+        self.ensure_one()
+        line_obj = self.env['stock.card.report.wizard.line']
+
+        def _create_lines(stock_card, location):
+            balance = 0.00
+            for line in stock_card:
+                if line.location_dest_id.id == location:
+                    balance = balance + line.product_uom_qty
+                    vals = self._prepare_line(line, 'in', balance, location)
+                    line_obj.create(vals)
+                elif line.location_id.id == location:
+                    balance = balance - line.product_uom_qty
+                    vals = self._prepare_line(line, 'out', balance, location)
+                    line_obj.create(vals)
+
+        stock_card = self.env['stock.move'].search([
```

Comment:
**Question:**
* Why not use stock.move.line?
* Make sure to sort date?
**Note:**
* Please be noted about the UoM conversion. You might need to convert into smallest UoM
* Do you already have Initial Balance, when user select start date after the first move date